lib/assets/javascripts/react_ujs_mount.js

```diff
@@ -42,50 +42,27 @@
       }
     },
 
-    // Get the constructor for a className
-    getConstructor: function(className) {
-      // Assume className is simple and can be found at top-level (window).
-      // Fallback to eval to handle cases like 'My.React.ComponentName'.
-      // Also, try to gracefully import Babel 6 style default exports
-      //
-      var constructor;
-
-      // Try to access the class globally first
-      constructor = window[className];
-
-      // If that didn't work, try eval
-      if (!constructor) {
-        constructor = eval.call(window, className);
-      }
-
-      // Lastly, if there is a default attribute try that
-      if (constructor && constructor.default) {
-        constructor = constructor.default;
-      }
-
-      return constructor;
-    },
-
     // Within `searchSelector`, find nodes which should have React components
     // inside them, and mount them with their props.
     mountComponents: function(searchSelector) {
       var nodes = window.ReactRailsUJS.findDOMNodes(searchSelector);
 
       for (var i = 0; i < nodes.length; ++i) {
         var node = nodes[i];
-        var className = node.getAttribute(window.ReactRailsUJS.CLASS_NAME_ATTR);
-        var constructor = this.getConstructor(className);
         var propsJson = node.getAttribute(window.ReactRailsUJS.PROPS_ATTR);
         var props = propsJson && JSON.parse(propsJson);
 
-        if (typeof(constructor) === "undefined") {
-          var message = "Cannot find component: '" + className + "'"
-          if (console && console.log) { console.log("%c[react-rails] %c" + message + " for element", "font-weight: bold", "", node) }
-          var error = new Error(message + ". Make sure your component is globally available to render.")
-          throw error
-        } else {
-          ReactDOM.render(React.createElement(constructor, props), node);
+        var className = node.getAttribute(window.ReactRailsUJS.CLASS_NAME_ATTR);
+        var constructor = window[className];
+
+        if (typeof constructor === "undefined") {
+          if (!props || !props.appFileName) {
```

Comment:
`props.appFileName` is a very Flexport-specific thing. At this point I'd probably inline the react-rails code into our repo. It's now pretty tightly coupled to our app. We're never going to merge this stuff upstream.

Reply:
Hmm 🤔 I might be wrong, we might be able to extract just the `react_ujs` code into our repo and leave the rest of `react-rails` code untouched